main.nf

```diff
@@ -268,114 +200,382 @@ ${summary.collect { k,v -> "            <dt>$k</dt><dd><samp>${v ?: '<span style
-=======
-    echo $manifest.pipelineVersion > v_pipeline.txt
-    echo $workflow.nextflow.version > v_nextflow.txt
     fastqc --version > v_fastqc.txt
-    multiqc --version > v_multiqc.txt
->>>>>>> TEMPLATE
+    fastp -v 2> v_fastp.txt
+    megahit --version > v_megahit.txt
+    metabat2 -h 2> v_metabat.txt || true
+
+    # fake checkm data setRoot so we can read checkm version number
+    chd=\$(readlink -f checkm_data)
+    printf "\$chd\\n\$chd\\n" | checkm data setRoot
+
+    checkm -h > v_checkm.txt
+    refinem -h > v_refinem.txt
     scrape_software_versions.py > software_versions_mqc.yaml
     """
 }
 
 
-<<<<<<< HEAD
 /*
  * STEP 1 - Read trimming and pre/post qc
  */
-process atropos {
+process fastqc_raw {
     tag "$name"
-    publishDir "${params.outdir}/atropos", mode: 'copy'
+    publishDir "${params.outdir}/fastqc", mode: 'copy'
 
     input:
-    set val(name), file(reads) from read_files_atropos
-    val adapter from params.three
-    val adapter_reverse from reverse_three
+    set val(name), file(reads) from read_files_fastqc
+
+    output:
+    file "*_fastqc.{zip,html}" into fastqc_results
+
+    script:
+    """
+    fastqc -t "${task.cpus}" -q $reads
+    """
+}
+
+
+process fastp {
+    tag "$name"
+    // publishDir "${params.outdir}/fastp", mode: 'copy'
+
+    input:
+    set val(name), file(reads) from read_files_fastp
+    val adapter from params.adapter_forward
+    val adapter_reverse from params.adapter_reverse
+    val qual from params.mean_quality
+    val trim_qual from params.trimming_quality
 
     output:
-    set val(name), file("${name}_trimmed_R{1,2}.fastq.gz") into trimmed_reads
-    file("${name}.report.txt") into atropos_report
+    set val(name), file("${name}_trimmed*.fastq.gz") into trimmed_reads
 
     script:
-    def single = reads instanceof Path
-    if ( !single ) {
+    if ( !params.singleEnd ) {
         """
-        atropos trim -a ${adapter} -A ${adapter_reverse} --op-order GACQW \
-        --trim-n -o "${name}_trimmed_R1.fastq.gz" -p "${name}_trimmed_R2.fastq.gz" \
-        --report-file "${name}.report.txt" --no-cache-adapters --stats both \
-        -pe1 "${reads[0]}" -pe2 "${reads[1]}"
+        fastp -w "${task.cpus}" -q "${qual}" --cut_by_quality5 \
+            --cut_by_quality3 --cut_mean_quality "${trim_qual}"\
+            --adapter_sequence=${adapter} --adapter_sequence_r2=${adapter_reverse} \
+            -i "${reads[0]}" -I "${reads[1]}" \
+            -o ${name}_trimmed_R1.fastq.gz -O ${name}_trimmed_R2.fastq.gz
         """
     }
     else {
         """
-        echo not implemented :-(
+        fastp -w "${task.cpus}" -q "${qual}" --cut_by_quality5 \
+            --cut_by_quality3 --cut_mean_quality "${trim_qual}"\
+            --adapter_sequence="${adapter}" --adapter_sequence_r2="${adapter_reverse}" \
+            -i ${reads} -o "${name}_trimmed.fastq.gz"
         """
     }
 }
+trimmed_reads.into{trimmed_reads_megahit; trimmed_reads_metabat; trimmed_reads_fastqc}
```

Comment:
You could also shift this into the process before by adding 

`set val(name), file("${name}_trimmed*.fastq.gz") into (trimmed_reads_megahit, trimmed_reads_metabat, trimmed_reads_fastqc`